src/ui/PrintableUri/PrintableUri.ts

```diff
@@ -113,25 +126,80 @@ export class PrintableUri extends Component {
     return link.element;
   }
 
-  private renderParentsXml(element: HTMLElement, parentsXml: string) {
-    const xmlDoc: XMLDocument = Utils.parseXml(parentsXml);
-    const parents = xmlDoc.getElementsByTagName('parent');
-    const tokens: HTMLElement[] = [];
-    const separators: HTMLElement[] = [];
-    for (let i = 0; i < parents.length; i++) {
-      if (i > 0) {
-        const separator = this.buildSeparator();
-        separators.push(separator);
-        element.appendChild(separator);
-      }
-      const parent = <Element>parents.item(i);
-      const token = this.buildHtmlToken(parent.getAttribute('name'), parent.getAttribute('uri'));
-      tokens.push(token);
+  private parseXmlParents(parentsXml: string): Element[] {
+    const elements = Utils.parseXml(parentsXml).getElementsByTagName('parent');
+    const parents: Element[] = [];
+    for (let i = 0; i < elements.length; i++) {
+      parents.push(elements.item(i));
+    }
+    return parents;
+  }
+
+  private renderParents(renderOptions: IPrintableUriRenderOptions) {
+    $$(this.element).empty();
+    const lastIndex = renderOptions.parents.length - 1;
+    const lastIndexBeforeLastPart = lastIndex - 1;
```

Comment:
Maybe `beforeLastIndex` would be a shorter name that keeps the meaning?

Reply:
Good idea, I'll change it.